.github/scripts/promote_branch.sh

```diff
@@ -120,7 +120,7 @@ mkdir -p ${releaseServiceCatalogDir}
 
 echo -e "---\nPromoting release-service-catalog ${SOURCE_BRANCH} to ${TARGET_BRANCH}\n---\n"
 
-git clone "git@github.com:$ORG/$REPO.git" ${releaseServiceCatalogDir}
+git clone "https://oauth2:$GITHUB_TOKEN@github.com/$ORG/$REPO.git" ${releaseServiceCatalogDir}
```

Comment:
I wonder if there's a risk that the token will be leaked in the log. It shouldn't be the case since this script doesn't have `set -x`, but it's probably a good idea to check anyway - by running it with a dummy value.

Reply:
+1

Reply:
I tested it previously and as far as I can remember it did not leak, but it worth it cross checking.

Reply:
Perfect, thanks! I can see the linked run and it doesn't show the clone command (because no `set -x`), but it does show the env vars and it's masked there: `GITHUB_TOKEN: ***`

It's something I'd expect of any CI tool, but unfortunately Tekton doesn't have anything like that.

Reply:
Yep, here is a link with `set -x` as well https://github.com/johnbieren/release-service-catalog/actions/runs/7574501989/job/20629013551
It seems if I turn actions back on, the past runs show up even if I have deleted branches they came from

Reply:
I can see it, thanks. Interesting.